tests/check_consistent.py

```diff
@@ -182,11 +189,19 @@ def check_metadata() -> None:
             for key in data.get("tool", {}).get(tool, {}):
                 assert key in tk, f"Unrecognised {tool} key {key} for {distribution}"
 
-        specified_stubtest_platforms = set(data.get("tool", {}).get("stubtest", {}).get("platforms", []))
+        tool_stubtest = data.get("tool", {}).get("stubtest", {})
+        specified_stubtest_platforms = set(tool_stubtest.get("platforms", []))
         assert (
             specified_stubtest_platforms <= supported_stubtest_platforms
         ), f"Unrecognised platforms specified: {supported_stubtest_platforms - specified_stubtest_platforms}"
 
+        # Check that only specified platforms install packages:
+        for supported_plat in supported_stubtest_platforms:
+            if supported_plat not in specified_stubtest_platforms:
+                assert not tool_stubtest.get(
+                    metadata_mapping[supported_plat]
+                ), f"Installing system deps for unspecified platform {supported_plat} for {distribution}"
```

Comment:
(black will almost certainly reformat this)

```suggestion
                assert metadata_mapping[supported_plat] not in tool_stubtest, f"Installing system deps for unspecified platform {supported_plat} for {distribution}"
```